vignettes/adrs.Rmd

```diff
@@ -0,0 +1,597 @@
+)
+```
+
+## Pre-processing of Input Records {#input}
+
+The first step involves company-specific pre-processing of records for the required input to the downstream parameter functions.
+Note that this could be needed multiple times (e.g. once for investigator and once for Independent Review Facility (IRF) records).
+It could even involve merging input data from other sources besides `RS`, such as `ADTR`.
+
+This step would include any required selection/derivation of `ADT` or applying any necessary partial date imputations,
+updating `AVAL` (e.g. this should be ordered from best to worst response), and setting analysis flag `ANL01FL`. Common options for
+`ANL01FL` would be to set null for invalid assessments or those occurring after next anti-cancer therapy, or to only flag assessments
+on or after after date of first treatment/randomization, or rules to cover the case when a patient has multiple observations per visit
+(e.g. by selecting worst value).
+Another consideration could be extra potential protocol-specific sources of Progressive Disease such as radiological assessments,
+which could be pre-processed here to create a PD record to feed downstream derivations.
+
+The below shows an example of a possible company-specific implementation of this step.
+
+### Select Overall Response Records and Set Parameter Details
+
+In this case we use the overall response records from `RS` from the investigator as our starting point. The parameter details such
+as `PARAMCD`, `PARAM` etc will always be company-specific, but an example is shown below so that you can trace through how these
+records feed into the other parameter derivations.
+
+```{r}
+adrs <- adrs %>%
+  filter(RSEVAL == "INVESTIGATOR" & RSTESTCD == "OVRLRESP") %>%
+  mutate(
+    PARAMCD = "OVR",
+    PARAM = "Overall Response by Investigator",
+    PARCAT1 = "Tumor Response",
+    PARCAT2 = "Investigator",
+    PARCAT3 = "Recist 1.1"
+  )
+```
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = vars(USUBJID, VISIT, RSTESTCD, RSEVAL, PARAMCD, PARAM, PARCAT1, PARCAT2, PARCAT3)
+)
+```
+
+### Partial Date Imputation and Deriving `ADT`, `ADTM`, `AVISIT` etc
+
+If your data collection allows for partial dates, you could apply a company-specific imputation rule at this stage
+when deriving `ADTM` and `ADT`. For this example, here we impute missing day to last possible date.
+
+```{r}
+adrs <- adrs %>%
+  derive_vars_dtm(
+    dtc = RSDTC,
+    new_vars_prefix = "A",
+    date_imputation = "last"
+  ) %>%
+  derive_vars_dtm_to_dt(vars(ADTM)) %>%
+  mutate(AVISIT = VISIT)
+```
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = vars(USUBJID, AVISIT, PARAMCD, PARAM, RSSTRESC, RSDTC, ADT, ADTM)
+)
+```
+
+### Derive `AVALC` and `AVAL`
+
+Here we populate `AVALC` and create the numeric version as `AVAL` (ordered from best to worst response). If you were
+working in different oncology indications you might have different possible response values here, such as `iCR`.
```

Comment:
I would not mention `iCR` because it could suggest that the admiralonco function can handle such a value. Maybe we should also mention that the `AVAL` values does not affect the derivations of the parameters.

Reply:
fixed